app/models/embedded_ansible_worker/runner.rb

```diff
@@ -93,11 +93,30 @@ def provider_uri_hash
   end
 
   def raise_role_notification(notification_type)
+    type_id = NotificationType.find_by(:name => notification_type).id
+    notifications = Notification.where(:notification_type_id => type_id)
+
+    if notifications.empty?
+      create_notification(notification_type)
+      return
+    end
+
+    create_notification(notification_type) unless notifications.any? do |n|
```

Comment:
We can probably handle the `if notifications.empty?` case here.  `notifications.any?` should be nearly as fast as `notifications.empty?` or the speed difference should not warrant the mental cost of an early return and extra conditional.  Of course, I assume it would work by just removing the entire `if notifications.empty?` condition.

Reply:
Unfortunately `[].any?` always returns false, so we can't remove the `empty?` check. I had a version of this where I combined the conditions, but this felt easier to read.

Reply:
Yeah, this one took some work to make it even a bit readable. Still looking for improvements if you've got anything.

I could also add comments if it would be helpful.